app/queries/event_procedures/by_year_query.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module EventProcedures
+  class ByYearQuery < ApplicationQuery
+    attr_reader :year, :relation
+
+    def initialize(year:, relation: EventProcedure)
+      @year = year
+      @relation = relation
+    end
+
+    def call
+      relation.where("EXTRACT(YEAR FROM date) = ?", year)
```

Comment:
please check if gpt is correct about using `relation.where(date: Date.new(year).all_year)`
Usually we avoid using raw SQL, but I see now problem here

Reply:
Done 01f3a136fe6e8e8869ea0dd2c2a7a6ea78319b53

Reply:
done 8bb46223a5c30a079cede74df662bd95574c9b12